tests/handlers/test_register.py

```diff
@@ -736,13 +757,15 @@ async def get_or_create_user(
 class RemoteAutoJoinTestCase(unittest.HomeserverTestCase):
     """Tests auto-join on remote rooms."""
 
-    def make_homeserver(self, reactor, clock):
+    def make_homeserver(self, reactor: MemoryReactor, clock: Clock) -> HomeServer:
         self.room_id = "!roomid:remotetest"
 
-        async def update_membership(*args, **kwargs):
+        async def update_membership(*args: Any, **kwargs: Any) -> None:
             pass
 
-        async def lookup_room_alias(*args, **kwargs):
+        async def lookup_room_alias(
+            *args: Any, **kwargs: Any
+        ) -> Tuple[RoomID, List[str]]:
             return RoomID.from_string(self.room_id), ["remotetest"]
```

Comment:
below here: is there any reason why we assign to `foo.side_effect` rather than `foo` directly? (Not that this's changed recently)

Reply:
It lets you see how it was called, on line 790 we have:

```
_, kwargs = self.room_member_handler.update_membership.call_args
```